core/services/chainlink/application.go

```diff
@@ -162,6 +176,9 @@ func NewApplication(config *orm.Config, ethClient eth.Client, advisoryLocker pos
 		shutdownSignal:           shutdownSignal,
 		balanceMonitor:           balanceMonitor,
 		explorerClient:           explorerClient,
+		// NOTE: Can keep things clean by putting more things in subservices
+		// instead of manually start/closing
```

Comment:
You didn't move the others over now to limit the PR scope?

Reply:
That was pretty much the thinking, yeah

Reply:
If we're laying down a new pattern here, its worth talking about using contexts instead of the manual stopping - i.e. ```type StartCloser interface { Start(ctx context.Context) }```. Curious what you guys think about that? The main win would be the subservices could easily spawn children and we'd get full tree cancellation for free + things like timeout support

Reply:
I'm for it, but think it should be a separate PR.